framework/meta/src/sc_upgrade/upgrade_common.rs

```diff
@@ -0,0 +1,166 @@
+use std::{fs, path::Path};
+
+use colored::Colorize;
+use ruplacer::{Console, DirectoryPatcher, Query, Settings};
+use toml::Value;
+
+use crate::CargoTomlContents;
+
+use super::{upgrade_versions::FRAMEWORK_CRATE_NAMES, version_req::VersionReq};
+
+/// Uses ruplacer.
+pub(crate) fn replace_in_files(sc_crate_path: &Path, file_type: &str, queries: &[Query]) {
+    let console = Console::default();
+    let settings = Settings {
+        selected_file_types: vec![file_type.to_string()],
+        ..Default::default()
+    };
+    let mut directory_patcher = DirectoryPatcher::new(&console, sc_crate_path, &settings);
+    for query in queries {
+        directory_patcher.run(query).expect("replace failed");
+    }
+}
+
+/// Regex was not needed yet, add if it becomes necessary.
+pub(crate) fn rename_files(path: &Path, patterns: &[(&str, &str)]) {
+    if path.is_file() {
+        if let Some(file_name) = path.file_name() {
+            if let Some(file_name_str) = file_name.to_str() {
+                for &(replace_pattern, replace_with) in patterns {
+                    if file_name_str.contains(replace_pattern) {
+                        let replaced_file_name =
+                            file_name_str.replace(replace_pattern, replace_with);
+                        let replaced_path = path.parent().unwrap().join(replaced_file_name);
+                        println!(
+                            "Renaming {} -> {}",
+                            path.display().to_string().red().strikethrough(),
+                            replaced_path.as_path().display().to_string().green(),
+                        );
+                        fs::rename(path, replaced_path).expect("failed to rename file");
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Less nesting please. The body of the `if`s should each be a separate function, using early exit conditions instead of deeply nested `if` statements.

Reply:
Fixed by creating smaller functions. Nice catch.